src/main/java/com/xatkit/core/recognition/processor/PerspectiveapiInterface.java

```diff
@@ -0,0 +1,281 @@
+         */
+        PROFANITY_EXPERIMENTAL,
+        /**
+         * Describes an intention to inflict pain, injury, or violence against an individual or group.
+         */
+        THREAT,
+        /**
+         * Describes an intention to inflict pain, injury, or violence against an individual or group.
+         */
+        THREAT_EXPERIMENTAL,
+        /**
+         * Contains references to sexual acts, body parts, or other lewd content.
+         */
+        SEXUALLY_EXPLICIT,
+        /**
+         * Pickup lines, complimenting appearance, subtle sexual innuendos, etc.
+         */
+        FLIRTATION,
+        /* New York Times models below */
+        /**
+         * Attack on the author of an article or post.
+         */
+        ATTACK_ON_AUTHOR,
+        /**
+         * Attack on fellow commenter.
+         */
+        ATTACK_ON_COMMENTER,
+        /**
+         * Difficult to understand, nonsensical.
+         */
+        INCOHERENT,
+        /**
+         * Intending to provoke or inflame.
+         */
+        INFLAMMATORY,
+        /**
+         * Overall measure of the likelihood for the comment to be rejected according to the NYT's moderation.
+         */
+        LIKELY_TO_REJECT,
+        /**
+         * Obscene or vulgar language such as cursing.
+         */
+        OBSCENE,
+        /**
+         * Irrelevant and unsolicited commercial content.
+         */
+        SPAM,
+        /**
+         * Trivial or short comments.
+         */
+        UNSUBSTANTIAL;
+    }
+
+    private String apiKey;
+    private String commentText;
+    private String commentType = "PLAIN_TEXT";
+    private AttributeType[] requestedAttributes;
+    private String language;
+    private Boolean doNotStore;
+    private String clientToken;
+    private String sessionId;
+    private JSONObject request;
+    private Map<String, AttributeType[]> languageAttributes;
+    private String attributesPrefix;
+
+    /**
+     * Constructor
+     *
+     * @param apiKey           the api key
+     * @param language         the language
+     * @param doNotStore       if true then data will not be stored to PerspectiveAPI. Default is false
+     * @param clientToken      the client token
+     * @param sessionId        the session id
+     * @param attributesPrefix the attributes prefix for the keys of the map they will be stored in
+     */
+    PerspectiveapiInterface(@NonNull String apiKey, String language, Boolean doNotStore, String clientToken,
+                            String sessionId, String attributesPrefix) {
+
+        this.apiKey = apiKey;
+        this.language = language != null ? language : "en";
+        this.doNotStore = doNotStore != null ? doNotStore : false;
+        this.clientToken = clientToken;
+        this.sessionId = sessionId;
+        this.attributesPrefix = attributesPrefix;
+        languageAttributes = new HashMap<>();
+        languageAttributes.put("en", new AttributeType[] {
```

Comment:
To discuss: create a `LanguageAttributes` class with a method like `List<AttributeType> getAttributesFor(String language)`

Reply:
Could be useful in a testing context (avoid to expose the class through a getter)